tests/manage/mcg/test_noobaa_secret.py

```diff
@@ -0,0 +1,332 @@
+        uls_name = list(cloud_uls_factory({cloud: [uls_tup]})["aws"])[0]
+        logger.info(f"ULS dict: {type(uls_name)}")
+        second_bs_name = create_unique_resource_name(
+            resource_description="backingstore",
+            resource_type=cloud.lower(),
+        )
+        bs_data = templating.load_yaml(constants.MCG_BACKINGSTORE_YAML)
+        bs_data["metadata"]["name"] = second_bs_name
+        bs_data["metadata"]["namespace"] = config.ENV_DATA["cluster_namespace"]
+        bs_data["spec"] = {
+            "type": "aws-s3",
+            "awsS3": {
+                "targetBucket": uls_name,
+                "region": "eu-central-1",
+                "secret": {
+                    "name": aws_client.secret.name,
+                    "namespace": bs_data["metadata"]["namespace"],
+                },
+            },
+        }
+        second_bs_obj = create_resource(**bs_data)
+        teardown_factory(second_bs_obj)
+
+        # Check if the duplicate secrets are allowed
+        first_bs_dict = OCP(
+            namespace=config.ENV_DATA["cluster_namespace"], kind="backingstore"
+        ).get(resource_name=first_bs_obj.name)
+        second_bs_dict = OCP(
+            namespace=config.ENV_DATA["cluster_namespace"], kind="backingstore"
+        ).get(resource_name=second_bs_name)
+        assert (
+            first_bs_dict["spec"]["awsS3"]["secret"]["name"]
+            == second_bs_dict["spec"]["awsS3"]["secret"]["name"]
+        ), "Backingstores are not referring to the same secrets when secrets with duplicate credentials are created!!"
+        logger.info(
+            "Duplicate secrets are not allowed! only the first secret is being referred"
+        )
+
+        # Modify the secret credentials to wrong one and see if the backingstores get rejected
+        first_secret_name = first_bs_dict["spec"]["awsS3"]["secret"]["name"]
+        wrong_access_key_patch = {"data": {"AWS_ACCESS_KEY_ID": "d3JvbmdhY2Nlc3NrZXk="}}
+        OCP(namespace=config.ENV_DATA["cluster_namespace"], kind="secret").patch(
+            resource_name=first_secret_name,
+            params=json.dumps(wrong_access_key_patch),
+            format_type="merge",
+        )
+        logger.info("Patched wrong access key!")
+        assert OCP(
+            namespace=config.ENV_DATA["cluster_namespace"], kind="backingstore"
+        ).wait_for_resource(
+            resource_name=first_bs_obj.name,
+            condition="Creating",
+            column="PHASE",
+        ), "Backingstores are not getting reconciled after changing linked secret credentials!"
+        logger.info("Backingstores getting reconciled!")
+
+    def test_noobaa_secret_deletion_m1(
+        self, backingstore_factory, teardown_factory, mcg_obj, cleanup
+    ):
+        """
+        Objectives of this test is:
+            1) create the secret using AWS credentials first
+            2) create a backingstore using CLI and passing the secret name
```

Comment:
No its correct only. Because under the hood, this line of code `backingstore_factory(
            method="cli", uls_dict={"aws": [(1, "eu-central-1")]}
        )[0]`  actually creates secret first and then uses the same secret to create backingstores. so it does the job as mentioned in docstrings!